### ISSUE TITLE
Opening a new terminal windows does not restore the previous window's size and position on the screen

### ISSUE THREAD

**Issue by Poopooracoocoo** [microsoft/terminal#12633]
### Windows Terminal version

1.13.10395.0

### Windows build number

10.0.19044

### Steps to reproduce

1. Open Terminal
2. Resize and reposition window
3. Close window
4. Open Terminal again

### Expected Behavior

Terminal would open the window in the same position and size

### Actual Behavior

The position and size are reset.

Note: I am not referring to session restore on start-up like you'd find in a web browser that can be enabled optionally. I'm referring to regular behaviour that people expect on desktop operating systems. And in particular, this makes it easier to use on smaller screens such as laptops.

**Comment by Rosefield** [comment]
Do you have window layout saving enabled? In Settings -> Startup -> "When Terminal starts" select the "Open windows from a previous session" option.

![image](https://user-images.githubusercontent.com/6185249/156939280-c487a790-58ad-4142-99aa-400d24443c95.png)


https://docs.microsoft.com/en-us/windows/terminal/customize-settings/startup#behavior-when-starting-a-new-terminal-session

**Comment by Poopooracoocoo** [comment]
> Do you have window layout saving enabled? In Settings -> Startup -> "When Terminal starts" select the "Open windows from a previous session" option.


I don't want that, nor should I need that enabled to have this.

**Comment by zadjii-msft** [comment]
Alright so this is a new feature request - a setting to use the `initialPosition`, `initialSize` in state.json instead of the ones in the settings file. I suppose that could be a checkbox within the Launch Size expander. Ah, but then we've got the trick that we'd want one checkbox for "use previous position" and another for "use previous size". (admittedly, initialPosition isn't in the SUI yet, #9075). Then there's also the complicating factor that enabling "Open windows from a previous session" would spooky-action-at-a-distance check and disable those boxes. 

So then it makes more sense to put this as another radio option in "When Terminal starts...". Or like, a checkbox under the "Open a tab with the default profile", that only ~reveals~enables itself when that radio button is selected? (hiding itself would be terrible discoverability)

**Comment by Poopooracoocoo** [comment]
An option doesn't really make sense to me. I don't really want anything complicated. I just want the default windowing behaviour I've come to expect from other apps. 🤷

**Comment by zadjii-msft** [comment]
We already have some other settings to control [where the window is positioned](https://docs.microsoft.com/en-us/windows/terminal/customize-settings/startup#launch-position) and [how big it is](https://docs.microsoft.com/en-us/windows/terminal/customize-settings/startup#launch-size), so already we're off the rails of "other apps". I'm of the strong opinion that user preference is king - I'd love to give everyone as many knobs to control the Terminal as possible, to make it work exactly like they'd like.

**Comment by Poopooracoocoo** [comment]
Good defaults are always nice tho :)

**Comment by SteveALee** [comment]
I agree, it should just work  its standard windows behaviour. Should respect the monitor it was on as well. Let people override if required but please follow principle of least surprises.

**Comment by DHowett** [comment]
Discussion outcome: This probably needs a miniature spec (AKA: we agree on it in the issue, and check in a quick markdown document about it) as the team couldn't come to consensus on whether this is a `firstWindowPreference` or a `launchPosition`, and whether this is part of or not a part of the session restoration _code_.

**Comment by SteveALee** [comment]
Thanks for bringing the outcome here!

**Comment by Stanzilla** [comment]
Hah just came here to report this, I was thinking of two suggestions:

1. Option to always spawn at a set of coordinates OR at the center of the screen
2. Remember position properly

**Comment by zadjii-msft** [comment]
FWIW:

[Option to always spawn at a set of coordinates](https://docs.microsoft.com/en-us/windows/terminal/customize-settings/startup#launch-position)  OR [at the center of the screen](https://docs.microsoft.com/en-us/windows/terminal/customize-settings/startup#center-on-launch) are already settings. 

Unifying those two with bullet point 2 is the one that this issue is tracking - finding a good way of exposing that setting to the user both as a UI and in json is the hard part.

**Comment by Stanzilla** [comment]
> FWIW:
> 
> [Option to always spawn at a set of coordinates](https://docs.microsoft.com/en-us/windows/terminal/customize-settings/startup#launch-position) OR [at the center of the screen](https://docs.microsoft.com/en-us/windows/terminal/customize-settings/startup#center-on-launch) are already settings.
> 
> Unifying those two with bullet point 2 is the one that this issue is tracking - finding a good way of exposing that setting to the user both as a UI and in json is the hard part.

Oh my bad, sorry. I didn't see it in the UI options.

**Comment by Poopooracoocoo** [comment]
Any updates here?

**Comment by zadjii-msft** [comment]
Nope. We'll make sure to update this thread when there is. In the meantime, might I recommend the Subscribe button?
![image](https://user-images.githubusercontent.com/18356694/91237459-5cbb0c80-e700-11ea-9347-b9b1ec2813b1.png)
That way you'll be notified of any updates to this thread, without needlessly pinging everyone on this thread ☺️

As it stands, this issue remains at:

> Discussion outcome: This probably needs a miniature spec (AKA: we agree on it in the issue, and check in a quick markdown document about it) as the team couldn't come to consensus on whether this is a `firstWindowPreference` or a `launchPosition`, and whether this is part of or not a part of the session restoration _code_.

If someone wants to sort out how exactly this should be implemented, we'd be more than happy to review a writeup ☺️

**Comment by lhecker** [comment]
> If someone wants to sort out how exactly this should be implemented, we'd be more than happy to review a writeup ☺️

FWIW, and when it comes to the implementation specifically, I believe this should use the [`GetWindowPlacement`](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getwindowplacement) and [`SetWindowPlacement`](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowplacement) APIs. Those have been in use by other HWND applications for this purpose since forever.

**Comment by Gavitron** [comment]
Is this still an open bug?  I'm unable to get `wt` to open on any monitor but the primary, without hard-coding an absolute position.

Windows Version   10.0.19045 Build 19045
Windows Terminal Version 1.17.11461.0

**settings.json:**
```
    "initialPosition": ",",
    "launchMode": "default",
    "firstWindowPreference": "defaultProfile",
    "centerOnLaunch": false,
```

### Steps to reproduce:  

1. Start windows terminal from shortcut on desktop of monitor two.
   ( Shortcut points to `%LOCALAPPDATA%\Microsoft\WindowsApps\wt.exe`)
2. terminal opens on monitor one.
3. drag window to monitor two. 
4. Close window via any method. (tested all of:  window-close widget, tab-close widget, alt+F4. typing "exit" in terminal)
5. Start windows terminal from shortcut on monitor two.

### Expected behaviour:
terminal opens on monitor two, with previous state

### Observed behaviour:
terminal opens on monitor one in a "default" state.  (though not one explicitly set)

note:  as alluded to above, I worked around this bug by hard-coding a window position, but now if I start a second terminal instance, it perfectly overlaps the first, instead of being offset by ~50px as usually happens when letting Windows decide.

**Comment by pro4tlzz** [comment]
> Is this still an open bug? I'm unable to get `wt` to open on any monitor but the primary, without hard-coding an absolute position.
> 
> Windows Version 10.0.19045 Build 19045 Windows Terminal Version 1.17.11461.0
> 
> **settings.json:**
> 
> ```
>     "initialPosition": ",",
>     "launchMode": "default",
>     "firstWindowPreference": "defaultProfile",
>     "centerOnLaunch": false,
> ```
> 
> ### Steps to reproduce:
> 1. Start windows terminal from shortcut on desktop of monitor two.
>    ( Shortcut points to `%LOCALAPPDATA%\Microsoft\WindowsApps\wt.exe`)
> 2. terminal opens on monitor one.
> 3. drag window to monitor two.
> 4. Close window via any method. (tested all of:  window-close widget, tab-close widget, alt+F4. typing "exit" in terminal)
> 5. Start windows terminal from shortcut on monitor two.
> 
> ### Expected behaviour:
> terminal opens on monitor two, with previous state
> 
> ### Observed behaviour:
> terminal opens on monitor one in a "default" state. (though not one explicitly set)
> 
> note: as alluded to above, I worked around this bug by hard-coding a window position, but now if I start a second terminal instance, it perfectly overlaps the first, instead of being offset by ~50px as usually happens when letting Windows decide.

if Microsoft wants to win people over they should fix this properly

**Comment by ChristianGruen** [comment]
> if Microsoft wants to win people over they should fix this properly

Yes. Please. Some people say MS doesn’t need to win people, unfortunately.

**Comment by zadjii-msft** [comment]
I'm pretty sure this has been fixed for some time, if you use "Open windows from a previous session" instead of "Open a tab with the default profile". I think 1.25 also introduced the ability to either restore just the window layout, or the layout & buffer contents (whereas in 1.24 it's always layout & contents)

**Comment by DHowett** [comment]
@zadjii-msft nah this is still "remember where i last closed a window, open the next window there-ish" with no session machinery at all
